### PR TITLE
chore: set VersionTLS12 as min default TLS version

### LINKED ISSUE #4193: KEDA should use TLS12 or above as minimum TLS version
### Report

Current, KEDA allows all TLS versions

### Expected Behavior

Not allowed broken TLS versions

### Actual Behavior

allowed broken TLS versions

### PR DESCRIPTION
**Pull request by JorTurFer**
### Checklist

- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)* https://github.com/kedacore/keda-docs/pull/1058
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)* https://github.com/kedacore/charts/pull/388
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to #https://github.com/kedacore/keda/issues/3961

Fixes #4193

### CODE REVIEW

File: pkg/util/http.go [hunk 1]
```diff
@@ -19,17 +19,34 @@ package util
...
+	if found {
+		switch version {
+		case "TLS13":
+			minTLSVersion = tls.VersionTLS13
+		case "TLS12":
+			minTLSVersion = tls.VersionTLS12
+		case "TLS11":
+			minTLSVersion = tls.VersionTLS11
+		default:
+			minTLSVersion = tls.VersionTLS10
```
Comment (zroubalik) [inline pkg/util/http.go:47]: Shouldn't this be `minTLSVersion = tls.VersionTLS12`?

Comment (JorTurFer) [reply pkg/util/http.go:47]: I added it to not fail with wrong versions. I mean, if the env isn't set, we will use `VersionTLS12`. Maybe I can add the case TLS10 and set default to VersionTLS12 with a message, let me chagne it

Comment (JorTurFer) [reply pkg/util/http.go:47]: I added it to not fail with wrong versions. I mean, if the env isn't set, we will use `VersionTLS12`. Maybe I can add the case TLS10 and set default to VersionTLS12 with a message, let me change it

Comment (JorTurFer) [reply pkg/util/http.go:51]: Done, PTAL

**Review by zroubalik — COMMENTED** [review]
Looking good, would be nice to add some unit tests, at least for `util/http.go` 🙏

**Comment by JorTurFer** [comment]
> Looking good, would be nice to add some unit tests, at least for `util/http.go` 🙏

Sure, I'll refactor the code to extract the function and test it :)

**Comment by JorTurFer** [comment]
PTAL @zroubalik

File: pkg/util/http_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		envValue:        "TLS13",
+		expectedVersion: tls.VersionTLS13,
+	},
+	{
+		envSet:          false,
+		expectedVersion: tls.VersionTLS12,
+	},
+}
+
+func TestResolveMinTLSVersion(t *testing.T) {
```
Comment (zroubalik) [inline pkg/util/http_test.go:60]:
```suggestion
func TestResolveMinTLSVersion(t *testing.T) {
     defer os.Unsetenv("KEDA_HTTP_MIN_TLS_VERSION")
```

Comment (zroubalik) [reply pkg/util/http_test.go:60]: let's clean after the test

**Review by zroubalik — CHANGES_REQUESTED** [review]

**Review by zroubalik — APPROVED** [review]
LGTM, great job!

File: pkg/util/http_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     defer os.Unsetenv("KEDA_HTTP_MIN_TLS_VERSION")
+	for _, testData := range minTLSVersionTestDatas {
```
Comment (zroubalik) [inline pkg/util/http_test.go:62]:
```suggestion
        defer os.Unsetenv("KEDA_HTTP_MIN_TLS_VERSION")
	for _, testData := range minTLSVersionTestDatas {
```

Comment (zroubalik) [reply pkg/util/http_test.go:62]: sorry, wrong formatting

**Comment by JorTurFer** [comment]
f**k gh editor...
I have fixed format issues locally xD

**Pull request merged**